sqlmesh/core/engine_adapter/mssql.py

```diff
@@ -40,25 +42,53 @@ class MSSQLEngineAdapter(
 
     DIALECT: str = "tsql"
     FALSE_PREDICATE = exp.condition("1=2")
+    SUPPORTS_TUPLE_IN = False
+    TRAILING_SEMICOLON = True
 
-    def table_exists(self, table_name: TableName) -> bool:
-        """
-        Similar to Postgres, MsSql doesn't support describe so I'm using what
-        is used there and what the redshift cursor does to check if a table
-        exists. We don't use this directly in order for this to work as a base
-        class for other postgres.
+    def columns(
+        self,
+        table_name: TableName,
+        include_pseudo_columns: bool = True,
+    ) -> t.Dict[str, exp.DataType]:
+        """MsSql doesn't support describe so we query information_schema."""
 
-        Reference: https://github.com/aws/amazon-redshift-python-driver/blob/master/redshift_connector/cursor.py#L528-L553
-        """
         table = exp.to_table(table_name)
 
-        catalog_name = table.args.get("catalog") or "master"
+        catalog_name = table.catalog or "master"
+        sql = (
+            exp.select("column_name", "data_type", "character_maximum_length")
+            .from_(f"{catalog_name}.information_schema.columns")
+            .where(f"table_name = '{table.name}'")
+        )
+        database_name = table.db
+        if database_name:
+            sql = sql.where(f"table_schema = '{database_name}'")
+
+        self.execute(sql)
+
+        columns_raw = self.cursor.fetchall()
+
+        columns = [
+            (col[0], f"{col[1]}({col[2]})") if col[1] == "varchar" and col[2] else (col[0], col[1])
```

Comment:
Aren't there other data types that have precision bits like `CHAR` and `DECIMAL`?